sdk/typescript/src/domains/zeto.ts

```diff
@@ -156,4 +336,85 @@ export class ZetoInstance {
     });
     return this.paladin.pollForReceipt(txID, this.options.pollTimeout);
   }
+
+  async setERC20(from: string, data: ZetoSetERC20Params) {
+    const txID = await this.paladin.sendTransaction({
+      type: TransactionType.PUBLIC,
+      abi: zetoPublicAbi,
+      function: "setERC20",
+      to: this.address,
+      from,
+      data,
+    });
+    const result = await this.paladin.pollForReceipt(txID, POLL_TIMEOUT_MS);
+    if (result === undefined) {
+      throw new Error("Failed to set ERC20");
+    }
+    this.erc20 = data._erc20;
+    return result;
+  }
+
+  async deposit(from: string, data: ZetoDepositParams) {
+    // first approve the Zeto contract to draw the amount from our balance in the ERC20
+    const txID1 = await this.paladin.sendTransaction({
+      type: TransactionType.PUBLIC,
+      abi: erc20Abi,
+      function: "approve",
```

Comment:
It feels weird for the Paladin SDK to make an "approve" call to the ERC20. I think this should be handled separately by the user, as we shouldn't make assumptions about their ERC20 contract (only about Paladin contracts).

Reply:
moved it out of the SDK and into the example code